tls-client/main.cpp

```diff
@@ -132,6 +133,8 @@ class HelloHTTPS {
         mbedtls_x509_crt_free(&_cacert);
         mbedtls_ssl_free(&_ssl);
         mbedtls_ssl_config_free(&_ssl_conf);
+        _tcpsocket->close();
```

Comment:
My main reason for suggesting this change wasn't some effective gain on the robustness of the application, I merely prefer that solution because of the logical coherence of the code.